src/Microsoft.Azure.Batch.SoftwareEntitlement.Server/Controllers/SoftwareEntitlementsController.cs

```diff
@@ -154,6 +184,32 @@ private ObjectResult CreateMissingRequestError()
             return StatusCode(400, error);
         }
 
+        private ObjectResult CreateMissingHostIdError()
+        {
+            _logger.LogError("No hostId specified");
+
+            var error = new SoftwareEntitlementFailureResponse
+            {
+                Code = "EntitlementDenied",
+                Message = new ErrorMessage("Missing hostId value from software entitlement request.")
+            };
+
+            return StatusCode(400, error);
+        }
```

Comment:
There are enough of these that it's probably prudent to create a factory method that creates them all.

Something like:
```
private ObjectResult CreateEntitlementDeniedError(
    string errorMessage,
    string logMessage) ...
```
Each of these methods would then become a single line calling the factory.